### DISCUSSION TITLE
connections.disconnect() takes 160ms and how to solve the long time problem

### DISCUSSION THREAD

**Discussion by zyy20191** [milvus-io/milvus#19674] [Q&A and General discussion]
**This is how I connect：**
nickname = str(time.time()).split('.')[1]
connections.connect(
            alias=nickname,
            host=MilvusParam["host"],
            port=MilvusParam["port"]
        )
t_one = Collection(name="dialog_nlp", using=nickname)

**This is my way of disconnecting：**
connections.disconnect(nickname)

So I have a few questions:
1.Why does disconnecting take so long? Is there any optimization method?
2.How long does it take to disconnect if I don't disconnect myself?

**Comment by xiaofan-luan** [comment]
@czs007 @XuanYang-cn  any comments?

**Comment by yhmo** [comment]
@zyy20191 
How many connections are created at the same time? If there are lot of connections connect/disconnect at the same time, I think the time cost could be large.

The disconnect() method:
```
    def disconnect(self, alias: str):
        if not isinstance(alias, str):
            raise ConnectionConfigException(-1, ExceptionsMessage.AliasType % type(alias))

        if alias in self._connected_alias:
            self._connected_alias.pop(alias).close()
```

The **_connected_alias** is a dict, we can see the disconnect() method do nothing but just pick a rpc channel and close it. The close() method is managed by grpc lib.

**Reply by zyy20191** [reply]
just one connection. Create a connection -> perform a query -> disconnect.
So I don't understand why it takes so long.

**Reply by zyy20191** [reply]
Can you test how long you need? If you're fast, it's probably my problem

**Reply by zyy20191** [reply]
**This is how I test time：**
dt_1 = datetime.datetime.now()
connections.disconnect('default')
dt_2 = datetime.datetime.now()
print((dt_2 - dt_1).total_seconds())

**Reply by yhmo** [reply]
It costs 200ms on my machine. All the time is spent by grpc channel.close().

**Reply by yhmo** [reply]
"How long does it take to disconnect if I don't disconnect myself?"
This is controlled by grpc channel configuration: keepalive_timeout_ms, the default value is 20 second: https://github.com/grpc/grpc/blob/master/doc/keepalive.md

**Reply by zyy20191** [reply]
fine, it seems that there is no way to shorten this time.I'll do it in one connected way.
